### ISSUE TITLE
Tests don't include generated Avro Classes

### ISSUE THREAD

**Issue by jszmajda** [commercehub-oss/gradle-avro-plugin#36]
Using this `build.gradle`: https://gist.github.com/jszmajda/324419cf9277cbbacc37ba4632a8a0db

My tests have `Unresolved reference` errors when I refer to the generated classes. The classes seem to generate OK and IntelliJ can find the generated classes for hinting, etc.

What am I missing?

Thanks!

**Comment by eshepelyuk** [comment]
Do you mean your **Kotlin tests** do not see generated classes ?

**Comment by jszmajda** [comment]
Yes, sorry, my Kotlin tests do not see the generated classes

**Comment by eshepelyuk** [comment]
We solved this by manually adding generated classes to sourceSet in `build.gradle`

**Comment by jszmajda** [comment]
Can you share an example? I'm still kind of new to Gradle :)

**Comment by davidmc24** [comment]
Could you provide an example project that demonstrates the situation?  My belief is that, by default, the generated classes from Avro files placed in src/main/avro should be available on the classpath to any classpath that includes the result of the default Java plugin's main compilation.  Languages/plugins/source sets that don't do so automatically may require additional configuration to instruct them to do so, as suggested by @eshepelyuk.

**Comment by jszmajda** [comment]
@davidmc24 sure, shared my `build.gradle` at the top, I have avsc files that get compiled properly to `build/generated-main-avro-java` as java and are being compiled to classes in `out/production/classes`. I could put together a sample project if you like too?

**Comment by eshepelyuk** [comment]
@jszmajda I suppose smth like this gonna work, but I'm not sure

```
compileTestKotlin.source(generateAvro.outputs)
```

**Comment by jszmajda** [comment]
@eshepelyuk hm yeah that wasn't it but I'll play with it, thanks :)

**Comment by jszmajda** [comment]
OK! I worked it out, thanks so much for your help with the `sourceSets` thing, that was the missing link.

TL;DR I added this to my build.gradle:
```groovy
sourceSets {
    main.java.srcDirs += 'build/generated-main-avro-java'
}
```

I also created a trivial project demonstrating this setup: https://github.com/jszmajda/gradle-avro-kotlin-sample

Thanks again! 🎉

**Comment by davidmc24** [comment]
It looks like the authors of the Kotlin Gradle plugin decided that Kotlin compilation should come before Java compilation.  This results in the inability of Kotlin classes to resolve any Java classes that are compiled by the default Java compilation.

The being said, the Kotlin plugin makes an attempt to compile the Java files that the default Java compilation would, and this plugin is currently incompatible with that approach, barring work-around configuration.  It seems like the problem is that I'm currently adding the generation task's outputs to the JavaCompilation task's sources, where maybe I should instead be adding them as srcDirs to the sourceset.  I'm re-opening this issue to investigate whether that change will work well.

**Comment by davidmc24** [comment]
I think I've pushed changes that should make it so that this plugin is compatible with kotlin-gradle-plugin without any need for special build configuration.  With the new version, it should no longer be necessary to manually add the generated sources to the sourceSet.  Any chance either of you ( @jszmajda  or @eshepelyuk ) could take a look a the changes and/or test them?  Unless I hear back that it doesn't work well, I'll probably push out a new release sometime next week, as the likelihood of the changes causing problems seems low, and my tests seem to indicate that it at least improves the situation.

**Comment by eshepelyuk** [comment]
Unfortunately I can't help much, since we're not using Kotlin, we did smth similart with `Scala` + `Java` project

**Comment by davidmc24** [comment]
@eshepelyuk If you think a similar sort of integration might be useful for Scala support, please open an issue.

**Comment by jszmajda** [comment]
@davidmc24 Thanks! I'm not familiar enough with gradle to know how to base a dependency off a git repo directly. If you can point me in the right direction I'll happily test! (sorry!)

**Comment by davidmc24** [comment]
Released in 0.12.0.  If this still seems to require special config to work with the new version, please re-open this issue.
